HTTP authentication not working

Hello!

I'm running a private instance of webpagetest behind HTTP auth. It works through the browser. However, if I enter the same stuff in the command line:

```
$ webpagetest test https://www.lostmy.name/ --server http://local-wpt/ -g user -w pass
```

It doesn't work:

```
{
  "error": {
    "name": "WPTAPIError",
    "code": 401,
    "message": "Unauthorized"
  }
}
```

Any ideas? Is this a problem with the module, or with how I'm using it?

On it.

Tests are failing on a clean branch, though:

```
  1) Example WebPageTest for Specs Hits a Nock Server gets a sync test with results with perf test specs request then waits for test results object:
     Uncaught AssertionError: { [Error: Nock: No match for request GET http://foobar10.com/runtest.php?url=http%3A%2F%2Ftwitter.com%2Fmarcelduran&runs=3&fvonl == 0
      at test/specs-test.js:38:16
      at WebPageTest.testCallback (lib/webpagetest.js:350:14)
      at callbackYield (lib/webpagetest.js:138:14)
      at WebPageTest.apiCallback (lib/webpagetest.js:207:7)
      at OverriddenClientRequest.onError (lib/webpagetest.js:131:5)
      at node_modules/nock/lib/request_overrider.js:208:11

  2) Example WebPageTest for Sync Hits a Nock Server gets a sync test with results with custom median metric request then waits for test results object:
     Error: Nock: No match for request GET http://foobar2.com/runtest.php?url=http%3A%2F%2Ftwitter.com%2Fmarcelduran&runs=3&fvonly=1&medianMetric=TTFB&f=json&pingback=http%3A%2F%2F127.0.0.1%3A8001%2Ftestdone
      at RequestOverrider.end (node_modules/nock/lib/request_overrider.js:259:17)
      at OverriddenClientRequest.RequestOverrider.req.end (node_modules/nock/lib/request_overrider.js:167:7)
      at Object.exports.get (http.js:36:7)
      at WebPageTest.get (lib/webpagetest.js:80:19)
      at WebPageTest.api (lib/webpagetest.js:177:9)
      at WebPageTest.listener (lib/webpagetest.js:386:9)
      at emitListeningNT (net.js:1260:10)
```

--- a/lib/helper.js
+++ b/lib/helper.js
@@ -181,6 +181,7 @@ function normalizeServer(server) {
   return {
     protocol: server.protocol,
     hostname: server.hostname,
+    auth: server.auth,
     pathname: server.pathname,
     port: parseInt(server.port, 10) || (server.protocol === 'https:' ? 443 : 80)
   };

--- a/lib/webpagetest.js
+++ b/lib/webpagetest.js
@@ -58,20 +58,34 @@ var filenames = {
 // GET helper function
 function get(config, pathname, proxy, callback, encoding) {
   var protocol = (config.protocol === 'https:' ? https : http),
-      options = proxy ? {
-        host: proxy.split(':')[0],
-        port: proxy.split(':')[1],
-        path: config.protocol+ '//' +config.hostname+':' + config.port+pathname,
-        headers: {
-          Host: config.hostname
-        }
-      } :
-      {
-        path: pathname,
-        host: config.hostname,
-        port: config.port,
-        headers: {}
-      };
+      options;
+
+  if (proxy) {
+    var pathForProxy = config.protocol + '//';
+
+    if (config.auth) {
+      pathForProxy += config.auth + '@';
+    }
+
+    pathForProxy += config.hostname + ':' + config.port + pathname;
+
+    options = {
+      host: proxy.split(':')[0],
+      port: proxy.split(':')[1],
+      path: pathForProxy,
+      headers: {
+        Host: config.hostname
+      }
+    };
+  } else {
+    options = {
+      path: pathname,
+      host: config.hostname,
+      auth: config.auth,
+      port: config.port,
+      headers: {}
+    };
+  }
 
   if (encoding !== 'binary') {
     options.headers['accept-encoding'] = 'gzip,deflate';

--- a/test/edge-cases-test.js
+++ b/test/edge-cases-test.js
@@ -144,6 +144,7 @@ describe('Edge Cases of', function() {
     it('when getting a valid standard server uri returns server info object', function() {
       var server = helper.normalizeServer('example.com');
       assert.deepEqual(server, {
+        auth: null,
         protocol: 'http:',
         hostname: 'example.com',
         pathname: '/',
@@ -152,8 +153,9 @@ describe('Edge Cases of', function() {
     });
 
     it('when getting a valid full server uri returns server info object', function() {
-      var server = helper.normalizeServer('http://example.com:8000/foo');
+      var server = helper.normalizeServer('http://foo:bar@example.com:8000/foo');
       assert.deepEqual(server, {
+        auth: 'foo:bar',
         protocol: 'http:',
         hostname: 'example.com',
         pathname: '/foo',
@@ -164,6 +166,7 @@ describe('Edge Cases of', function() {
     it('when getting a https server uri returns server info object', function() {
       var server = helper.normalizeServer('https://example.com');
       assert.deepEqual(server, {
+        auth: null,
         protocol: 'https:',
         hostname: 'example.com',
         pathname: '/',

